Container Workload I

shubham3121 opened the pull request:
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

shubham3121 committed:
define classes to store image meta info
- create config class for storing dockerfile
- define a basic stash and service class for worker images
-  create a method to build image using docker python sdk

Co-authored-by: Yash Gorana <yash.gorana@hotmail.com>
Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>
Co-authored-by: Peter Chung <peterchung241@gmail.com>
Co-authored-by: Khoa Nguyen <dkn.work@protonmail.com>

rasswanth-s committed:
Add method to worker Image Service
- Submit DockerFiles
- List DockerFile
- Delete DockerFile
- Build Dockerfile

Co-authored-by: Yash Gorana <yash.gorana@hotmail.com>
Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>
Co-authored-by: Peter Chung <peterchung241@gmail.com>
Co-authored-by: Khoa Nguyen <dkn.work@protonmail.com>
Co-authored-by: Kien Dang <kiendang@users.noreply.github.com>
Co-authored-by: Shubham Gupta <shubhamgupta3121@gmail.com>

kiendang commented on packages/syft/src/syft/custom_worker/config.py:
```diff
@@ -95,3 +101,25 @@ def from_path(cls, path: Union[Path, str]) -> Self:
+
+    @classmethod
+    def from_str(cls, content: str) -> Self:
+        return cls(dockerfile=content)
```
This method is redundant? It's the same as `DockerWorkerConfig(content)`?

shubham3121 replied:
Yep agreed. We can remove it.

kiendang committed:
Minor noop cleanup

kiendang committed:
Fix notebook

shubham3121 committed:
import Self from typing_extension

kiendang committed:
Fix notebook typo

kiendang committed:
Remove redundant method

kiendang committed:
Merge branch 'dev' into container-workload-1

shubham3121 committed:
initial worker on worker pool
- Define data structures for Worker and Worker Pool
- Define WorkerPoolStash
- Define WorkerPoolService
- define some utils

khoaguin committed:
[syft/worker_image_service] delete the corresponding built image given full image tag

kiendang commented on packages/syft/src/syft/service/worker/worker_image_service.py:
```diff
@@ -0,0 +1,151 @@
+        name="delete",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def delete(
+        self, context: AuthedServiceContext, uid: UID, clear_image: bool = False
+    ) -> Union[SyftSuccess, SyftError]:
```
I think we should always remove the image with `delete`, getting rid of the `clear_image` arg altogether. When an user do `delete` that's probably what they mean. I don't see a use case for deleting the image entry from the stash but leaving the image there. Deleting the image entry but not the image itself would lead to further complications later on (`e.g.` dangling images).

khoaguin replied:
I agree @kiendang. Fixed the code accordingly

kiendang commented on notebooks/api/0.8/10-container-images.ipynb:
@khoaguin We should not use `docker` cli command in the notebook here (`!docker images`). The user is submitting the Dockerfile to be built on the server. They might not have `docker` locally.

khoaguin committed:
[syft/worker_image_service] making deleting built image a default behavior
remove `!docker` from api notebook

shubham3121 committed:
implement method to run container for a docker image
- add a new class to store container spawn status
- add new Worker status
- fix pool create API

Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>
Co-authored-by: Yash Gorana <yash.gorana@hotmail.com>
Co-authored-by: Peter Chung <peterchung241@gmail.com>
Co-authored-by: Khoa Nguyen <dkn.work@protonmail.com>
Co-authored-by: Kien Dang <kiendang@users.noreply.github.com>

rasswanth-s committed:
- Add a get_all functionality to workerpool service.py
- Testing worker pool creation with images

Co-authored-by: Yash Gorana <yash.gorana@hotmail.com>
Co-authored-by: rasswanth <43314053+rasswanth-s@users.noreply.github.com>
Co-authored-by: Peter Chung <peterchung241@gmail.com>
Co-authored-by: Khoa Nguyen <dkn.work@protonmail.com>
Co-authored-by: Kien Dang <kiendang@users.noreply.github.com>
Co-authored-by: Shubham Gupta <shubhamgupta3121@gmail.com>

shubham3121 committed:
Merge branch 'dev' into container-workload-1

rasswanth-s committed:
Fix Notebook Tests for Container Images
Ignored the container tests for syft python tests

Co-authored-by: Shubham Gupta <shubhamgupta3121@gmail.com>

shubham3121 committed:
skip container workerload notebook for k8s

shubham3121 committed:
add port to orchestra launch

shubham3121 committed:
fix unique attr for SyftWorkerImage

shubham3121 committed:
mount /var/run/docker.sock to worker in docker-compose.yml
- fix protocol_version.json
- fix syftimageworker delete api

shubham3121 committed:
Merge branch 'dev' into container-workload-1
